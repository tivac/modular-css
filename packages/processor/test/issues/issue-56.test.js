"use strict";

const dedent = require("dedent");
const namer  = require("@modular-css/test-utils/namer.js");
const Processor = require("../../processor.js");

describe("/issues", () => {
    describe("/56", () => {
        it("should prune rules that only compose, but leave them in the exports", async () => {
            const processor = new Processor({
                    namer,
                });
            
            const { exports } = await processor.string(
                "./packages/processor/test/specimens/issues/56.css",
                dedent(`
                    .booga { color: red }
                    .fooga { composes: booga }
                    .fooga:hover { color: blue }
                    .wooga { composes: booga }
                `)
            );

            expect(exports).toMatchSnapshot();

            const { css } = await processor.output();
            
            expect(css).toMatchSnapshot();
        });
    });
});
