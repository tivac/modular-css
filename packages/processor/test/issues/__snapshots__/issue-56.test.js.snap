// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/issues /56 should prune rules that only compose, but leave them in the exports 1`] = `
Object {
  "booga": Array [
    "mc_booga",
  ],
  "fooga": Array [
    "mc_booga",
    "mc_fooga",
  ],
  "wooga": Array [
    "mc_booga",
    "mc_wooga",
  ],
}
`;

exports[`/issues /56 should prune rules that only compose, but leave them in the exports 2`] = `
"/* packages/processor/test/specimens/issues/56.css */
.mc_booga { color: red }
.mc_fooga:hover { color: blue }"
`;
